src/Features/Core/Portable/UseSystemHashCode/Analyzer.cs

```diff
@@ -0,0 +1,352 @@
+            // Last statement has to be the return of it.
+            if (!(statements.First() is IVariableDeclarationGroupOperation varDeclStatement) ||
+                !(statements.Last() is IReturnOperation returnStatement))
+            {
+                return default;
+            }
+
+            var variables = varDeclStatement.GetDeclaredVariables();
+            if (variables.Length != 1 ||
+                varDeclStatement.Declarations.Length != 1)
+            {
+                return default;
+            }
+
+            var declaration = varDeclStatement.Declarations[0];
+            if (declaration.Declarators.Length != 1)
+            {
+                return default;
+            }
+
+            var declarator = declaration.Declarators[0];
+            if (declarator.Initializer == null ||
+                declarator.Initializer.Value == null)
+            {
+                return default;
+            }
+
+            var hashCodeVariable = declarator.Symbol;
+            if (!(IsLocalReference(returnStatement.ReturnedValue, hashCodeVariable)))
+            {
+                return default;
+            }
+
+            var hashedSymbols = ArrayBuilder<ISymbol>.GetInstance();
+
+            // Local declaration can be of the form:
+            //
+            //      // VS code gen
+            //      var hashCode = number;
+            //
+            // or
+            //
+            //      // ReSharper code gen
+            //      var hashCode = Hash(firstSymbol);
+
+            // Note: we pass in `seenHash: true` here because ReSharper may just initialize things
+            // like `var hashCode = intField`.  In this case, there won't be any specific hashing
+            // operations in the value that we have to look for.
+            var initializerValue = declarator.Initializer.Value;
+            if (!IsLiteralNumber(initializerValue) &&
+                !TryAddHashedSymbol(initializerValue, seenHash: true))
+            {
+                return default;
+            }
+
+            // Now check all the intermediary statements.  They all have to be of the form:
+            //
+            //      hashCode = (hashCode op constant) op Hash(member)
+            //
+            // Or recursively built out of that.  For example, in VB we sometimes generate:
+            //
+            //      hashCode = Hash((hashCode op constant) op Hash(member))
+            //
+            // So, after confirming we're assigning to our accumulator, we recursively break down
+            // the expression, looking for valid forms that only end up hashing a single field in
+            // some way.
+            for (var i = 1; i < statements.Length - 1; i++)
+            {
+                var statement = statements[i];
+                if (!(statement is IExpressionStatementOperation expressionStatement) ||
+                    !(expressionStatement.Operation is ISimpleAssignmentOperation simpleAssignment) ||
```

Comment:
`if (!(statement is IExpressionStatementOperation { Operation: ISimpleAssignmentOperation simpleAssignment }) ||`

since `expressionStatement` is only used once?

Reply:
i tried.  but these end up just looking too complex and cluttered for me :(

Reply:
Agreed....I'm not sure I can parse that either. :-)